xskillscore/core/types.py

```diff
@@ -0,0 +1,10 @@
+from __future__ import annotations
+
+from typing import List, Union
+
+import xarray as xr
+
+XArray = Union[xr.Dataset, xr.DataArray]
+# XArray = xr.Dataset | xr.DataArray # raises error during build: TypeError: unsupported operand type(s) for |: 'ABCMeta' and 'type'
+Dim = Union[str, List[str], None]
+# Dim = str | List[str] | None
```

Comment:
Didn’t work

Reply:
any idea why the commented out lines dont work?

Reply:
Random jumping in: My guess -- wrong python version. `|` isn't supported until python 3.10

Reply:
but I use `from __future__ import annotations` at the top

Reply:
I believe that adds general annotation support, but not support of `|` operator. I don't think it's possible to get that without having python 3.10. 

```python
from future import __annotations__

X = str | int
```
mypy raises an error for me